packages/ckeditor5-ui/src/arialiveannouncer.ts

```diff
@@ -166,58 +144,104 @@ export class AriaLiveAnnouncerView extends View {
+	declare public regionName: string;
 
 	/**
-	 * Current politeness level of the region.
+	 * DOM converter used to sanitize unsafe HTML passed to {@link #announce} method.
 	 */
-	declare public politeness: typeof AriaLiveAnnouncerPoliteness[ keyof typeof AriaLiveAnnouncerPoliteness ];
+	private _domConverter: DomConverter;
 
 	/**
-	 * A unique name of the region, usually associated with a specific editor feature or system.
+	 * Interval used to remove additions. It prevents accumulation of added nodes in region.
 	 */
-	declare public regionName: string;
+	private _pruneAnnouncementsInterval: ReturnType<typeof setInterval> | null;
 
-	constructor( locale: Locale, domConverter: DomConverter ) {
+	constructor( locale: Locale, editor: Editor ) {
 		super( locale );
 
 		const bind = this.bindTemplate;
 
 		this.set( 'regionName', '' );
-		this.set( 'content', '' );
-		this.set( 'isUnsafeHTML', false );
 		this.set( 'politeness', AriaLiveAnnouncerPoliteness.POLITE );
 
 		this.setTemplate( {
 			tag: 'div',
 			attributes: {
 				role: 'region',
 				'data-region': bind.to( 'regionName' ),
-				'aria-live': bind.to( 'politeness' )
-			}
+				'aria-live': bind.to( 'politeness' ),
+				'aria-relevant': 'additions'
+			},
+			children: [
+				{
+					tag: 'ul',
+					attributes: {
+						class: [
+							'ck',
+							'ck-aria-live-region-list'
+						]
+					}
+				}
+			]
 		} );
 
-		this.on<ObservableChangeEvent<string>>( 'change:content', ( evt, name, content ) => {
-			if ( this.isUnsafeHTML ) {
-				domConverter.setContentOf( this.element!, content );
-			} else {
-				this.element!.innerText = content;
+		editor.on( 'destroy', () => {
+			if ( this._pruneAnnouncementsInterval !== null ) {
+				clearInterval( this._pruneAnnouncementsInterval! );
+				this._pruneAnnouncementsInterval = null;
 			}
 		} );
+
+		this._domConverter = editor.data.htmlProcessor.domConverter;
+		this._pruneAnnouncementsInterval = setInterval( () => {
+			if ( this.element && this._listElement!.firstChild ) {
+				this._listElement!.firstChild!.remove();
+			}
+		}, 5000 );
+	}
+
+	/**
+	 * Appends new announcement to region.
+	 */
+	public announce( { announcement, isUnsafeHTML }: AriaLiveAppendContentAttributes ): void {
+		if ( !announcement.trim().length ) {
+			return;
+		}
+
+		const messageListItem = document.createElement( 'li' );
+
+		if ( isUnsafeHTML ) {
+			this._domConverter.setContentOf( messageListItem, announcement );
+		} else {
+			messageListItem.innerText = announcement;
+		}
+
+		this._listElement!.appendChild( messageListItem );
+	}
+
+	/**
+	 * Return current announcements list HTML element.
+	 */
+	private get _listElement(): HTMLElement | null {
+		return this.element!.querySelector( 'ul' )!;
 	}
 }
 
 type AriaLiveAnnouncerPolitenessValue = typeof AriaLiveAnnouncerPoliteness[ keyof typeof AriaLiveAnnouncerPoliteness ];
 
+type AriaLiveAppendContentAttributes = {
+	announcement: string;
+	isUnsafeHTML?: boolean;
```

Comment:
Let's get rid of this flag as it seems it won't be used anyway.

Reply:
It is used in destination branch https://github.com/cksource/ckeditor5-commercial/pull/6163/files#diff-c3e2cfc5f2b64e23fc4bb25cdb3e76e1615a85a1fd0fac616a71d2271cb1456cR491-R494